File: bdp-master-server/src/main/java/org/bdp/master/server/service/AppService.java
package org.bdp.master.server.service;

import org.bdp.master.server.domain.App;

import java.util.List;

public interface AppService {
    App findOne(Long id);

    App findOne(String name);

    void save(App app);

    void delete(Long id);

    List<App> findAll();

    void loadAll();
}

File: bdp-master-client/src/main/scala/org/bdp/master/client/service/AppService.scala
package org.bdp.master.client.service

import org.bdp.master.client.Constants._
import com.typesafe.scalalogging.LazyLogging
import org.bdp.master.client.domain.App
import org.bdp.master.client.util.{JsonDecoder, RedisClient}

object AppService extends LazyLogging {

  def getAppBy(id: Long): App = {
    JsonDecoder.decodeApp(RedisClient.get(s"$APP_KEYSPACE:$id"))
  }

  def getAppBy(name: String): App = {
    val key = RedisClient.get(s"$INDEX_PREFIX$APP_KEYSPACE:$name")
    JsonDecoder.decodeApp(RedisClient.get(key))
  }

}
